### PR TITLE
[ch70] As a user, I can see a Registration Success screen

### PR DESCRIPTION
**Pull request by Kittinske15**
https://app.clubhouse.io/cobda/story/70/as-a-user-i-can-see-a-registration-success-screen

## Why?

This **PR** is the entry point for the ``Registration Success`` screen. And accessible by clicking ``Submit`` button which is located in ``Sign Up`` screen.

## Changes

- Create ``Registration Success`` screen.
- Add route to ``Registration Success`` screen.

## Screenshot(s)

**Before**
image

**After**
Registration Success screen was created.
image

### CODE REVIEW

File: src/components/RegistrationSuccessForm/styles.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@import "../../styles/main.scss";
+
+.container {
+    min-height: 100vh;
+    @include flex-center();
```
Comment (LilBank) [inline src/components/RegistrationSuccessForm/styles.scss]: A similar comment here: https://github.com/Cobda/cobda-web/pull/7#discussion_r563290999

File: src/components/RegistrationSuccessForm/styles.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+.success-form {
```
Comment (LilBank) [inline src/components/RegistrationSuccessForm/styles.scss]: According to the comment above, what do you think about naming it `sign-up-success`?

File: src/components/RegistrationSuccessForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import "./styles.scss";
+
+export default function RegistrationSuccessForm() {
```
Comment (LilBank) [inline src/components/RegistrationSuccessForm/index.tsx]: IMO, I'm comfortable with the word **Sign up** and **Sign in**. Should we stick we these words? @kimvcp @Raksani

File: src/components/Navbar/styles.scss [hunk 4]
```diff
@@ -15,4 +15,21 @@
...
+}
+
+.link-container {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  &__right {
+    position: absolute;
+    right: 1rem;
```
Comment (LilBank) [inline src/components/Navbar/styles.scss]: So we don't have to big brain time.
```suggestion
    right: rem(16px);
```

File: src/screens/RegistrationSuccess/index.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import RegistrationSuccessForm from '../../components/RegistrationSuccessForm';
+import "./styles.scss";
+
+export default function RegistrationSuccess() {
+  return (
+    <>
+      <div className="container">
```
Comment (LilBank) [inline src/screens/RegistrationSuccess/index.tsx]: This css class name relates to the `RegistrationSuccessForm` comment.

File: src/components/SignUpForm/styles.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    @include flex-center();
+}
+
+@mixin button-bg($background) {
+  background-color: $background;
+  margin-top: 16px;
+  &:hover {
+    background-color: darken($background, 8%);
+  }
+}
```
Comment (LilBank) [inline src/components/SignUpForm/styles.scss]: We don't create a `@mixin` within the component and use it 😅 Think of it as a public method from a utility class.

**Review by LilBank — CHANGES_REQUESTED** [review]

**Comment by LilBank** [comment]
Related comment about PR description: https://github.com/Cobda/cobda-web/pull/7#issuecomment-766347927

**Comment by kimvcp** [comment]
Please use this branch naming convention specified in https://github.com/Cobda/cobda-web/pull/7#issuecomment-766367669

File: src/components/Navbar/index.tsx [hunk 7]
```diff
@@ -1,6 +1,14 @@
...
-  return <nav className="navbar-container">COBDA</nav>;
+  return (
+    <nav className="navbar-container">
+      COBDA
+      <Link to="/sign-up" className="link-container link-container__right">
+        Sign Up
+      </Link>
+    </nav>
+  );
 }
```
Comment (Raksani) [inline src/components/Navbar/index.tsx]: Please add new line 
```suggestion
}

```

File: src/components/RegistrationSuccessForm/index.tsx (hunk 3 above)
Comment (Raksani) [reply src/components/RegistrationSuccessForm/index.tsx]: I'd like it to be 'SignUpSuccess[screen? modal?]' without the word 'form' because there's no form for users to fill, only just to show the progress (failed/success) ☺️

File: src/components/SignUpForm/styles.scss [hunk 8]
```diff
@@ -0,0 +1,37 @@
+@import "../../styles/main.scss";
```
Comment (Raksani) [inline src/components/SignUpForm/styles.scss]: Actually, it has to be with single code. 
```suggestion
@import '../../styles/main.scss';
```

File: src/components/SignUpForm/styles.scss [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+.container {
+    min-height: 100vh;
+    @include flex-center();
+}
+
+.signUp-form {
```
Comment (Raksani) [inline src/components/SignUpForm/styles.scss]: I think `sign-up-form` would be better.

File: src/components/SignUpForm/styles.scss [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    flex-direction: column;
+    
+    background-color: $background;
+  
+    @include flex-center();
+}
+
+@mixin button-bg($background) {
+  background-color: $background;
+  margin-top: 16px;
```
Comment (Raksani) [inline src/components/SignUpForm/styles.scss]: Convert to rem.

**Review by Raksani — CHANGES_REQUESTED** [review]

File: src/components/Navbar/styles.scss [hunk 11]
```diff
@@ -0,0 +1,35 @@
+@import "../../styles/main.scss";
+
+@mixin sticky() {
+  top: 0;
+  width: 100%;
+  position: fixed;
+}
+
+.navbar-container {
```
Comment (LilBank) [inline src/components/Navbar/styles.scss]: Please recheck this file after the rebase.

File: src/components/SignUpSuccessForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+const SignUpSuccessForm = () => {
+  return (
+    <section className="sign-up-success-form">
+      <header className="sign-up-success-form__header">Sample Picture</header>
```
Comment (LilBank) [inline src/components/SignUpSuccessForm/index.tsx]: For `header` tag, I think it would be better to use it with `h1, h2, h3, ..`. Let's just use `div` for this. What do you think?
```suggestion
      <div className="sign-up-success-form__image">Sample Picture</header>
```

File: src/components/SignUpSuccessForm/index.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+      <div className="sign-up-success-form__body">
+        <h2>Registration Success</h2>
+      </div>
+      <footer className="sign-up-success-form__footer">
+        <div className="form__actionable">
+          <button className="form__button">Continue</button>
+        </div>
+      </footer>
```
Comment (LilBank) [inline src/components/SignUpSuccessForm/index.tsx]: Since there's no `form` so I don't think we need to this. How about we make this button a `sign-up-success-form` element?
```suggestion
      <div className="sign-up-success-form__actionable">
        <button className="sign-up-success__button">Continue</button>
      </div>
```

File: src/styles/components/_sign-up-success-form.scss [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+.sign-up-success-form {
+  display: flex;
+  flex-direction: column;
+  background-color: $background;
+
+  &__header {
+    padding: rem(16px 0);
+    margin-bottom: rem(24px);
+
+    @include flex-center();
```
Comment (LilBank) [inline src/styles/components/_sign-up-success-form.scss]: Is there any difference between this?
```suggestion
    display: flex;
    justify-content: center;
```

File: src/components/SignUpSuccessForm/index.tsx [hunk 15]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+const SignUpSuccessForm = () => {
+  return (
+    <section className="sign-up-success-form">
+      <header className="sign-up-success-form__header">Sample Picture</header>
+      <div className="sign-up-success-form__body">
+        <h2>Registration Success</h2>
+      </div>
```
Comment (LilBank) [inline src/components/SignUpSuccessForm/index.tsx]: I think this suits better with `header` though since there's `h2` 🤔
```suggestion
      <header className="sign-up-success-form__header">
        <h2>Registration Success</h2>
      </header>
```

**Review by LilBank — CHANGES_REQUESTED** [review]

File: src/components/Form/index.tsx [hunk 16]
```diff
@@ -36,7 +36,7 @@ const Form = ({ title }: Form) => {
         <input type="text" className="form__input" />
       </div>
       <div className="form__actionable">
-        <Link href="/registration-success">
+        <Link href="/sign-up-success">
```
Comment (kimvcp) [inline src/components/Form/index.tsx]: There will surely be other logics rather than just navigation, I would consider using `Next Router` instead.

File: src/components/SignUpSuccessForm/index.tsx (hunk 12 above)
Comment (Kittinske15) [reply src/components/SignUpSuccessForm/index.tsx]: I agree with you because ``header`` tag should be kind of header for an article but for this one just using ``div`` tag to represent the image inside.

File: src/components/Form/index.tsx [hunk 17]
```diff
@@ -1,11 +1,19 @@
...
 interface Form {
   title: string
+  href: string
 }
 
-const Form = ({ title }: Form) => {
+const Form = ({ title, href }: Form) => {
+  const router = useRouter()
+
+  const handleFormSubmit = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (LilBank) [inline src/components/Form/index.tsx]: How about this?
```suggestion
  const onButtonSubmitClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```

File: src/components/SignUpSuccessForm/index.tsx [hunk 18]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+
+const SignUpSuccessForm = () => {
+  return (
+    <section className="sign-up-success-form">
+      <div className="sign-up-success-form__image">Sample Picture</div>
```
Comment (LilBank) [inline src/components/SignUpSuccessForm/index.tsx]: Can you set a this element a width and height for mockup?

File: src/components/SignUpSuccessForm/index.tsx [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+      <header className="sign-up-success-form__header">
+        <h2>Registration Success</h2>
+      </header>
+      <footer className="sign-up-success-form__footer">
+        <div className="sign-up-success-form__actionable">
+          <button className="sign-up-success-form__button">Continue</button>
+        </div>
+      </footer>
```
Comment (LilBank) [inline src/components/SignUpSuccessForm/index.tsx]: I think we don't need the footer though whereas we can add properties in `__actionable`? 🤔

File: src/styles/components/_form.scss [hunk 20]
```diff
@@ -32,11 +32,6 @@
     margin: 0;
   }
 
-  &__actionable {
-    display: flex;
-    justify-content: center;
-  }
```
Comment (LilBank) [inline src/styles/components/_form.scss:38]: Why is this removed?

File: src/styles/components/_sign-up-success-form.scss [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+
+  &__footer {
+    margin-bottom: rem(24px);
+  }
+
+  &__actionable {
+    display: flex;
+    justify-content: center;
+  }
+
```
Comment (LilBank) [inline src/styles/components/_sign-up-success-form.scss]: How about a little rework on here? 🤓

```suggestion
.sign-up-success-form {
  display: flex;
  flex-direction: column;
  background-color: $background;
  
  &__image,
  &__header,
  &__actionable {
    display: flex;
    justify-content: center;
    
    &:not(:last-child) {
      margin-bottom: rem(24px);
    }
  }

  &__image {
    padding: rem(16px 0);
    margin-bottom: rem(24px);
  }
}
```

File: src/styles/components/_sign-up-success-form.scss [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+  &__button {
+    color: $light;
+    text-decoration: none;
+    margin: 0;
+    padding: rem(5px 10px);
+    border-radius: rem(3px);
+    background-color: $shuttle-gray;
+
+    &:hover {
+      background-color: darken($background, 8%);
```
Comment (LilBank) [inline src/styles/components/_sign-up-success-form.scss]: Can we rename `$background` to it's actual color for better meaning?

**Review by LilBank — CHANGES_REQUESTED** [review]

File: src/styles/components/_form.scss (hunk 20 above)
Comment (Kittinske15) [reply src/styles/components/_form.scss:38]: My bad, I delete it in the wrong file

File: src/styles/components/_sign-up-success-form.scss [hunk 23]
```diff
@@ -0,0 +1,41 @@
+@import '../abstracts/_abstracts-dir';
+
+.sign-up-success-form {
+  display: flex;
+  flex-direction: column;
+  padding: rem(16px 0);
+  background-color: $background;
+
+  &__box,
```
Comment (LilBank) [inline src/styles/components/_sign-up-success-form.scss]: Should we rename this to `__image-container` for better meaning?

File: src/styles/components/_sign-up-success-form.scss [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+      margin-bottom: rem(24px);
+    }
+  }
+
+  &__image {
+    height: rem(192px);
+    width: rem(192px);
+    align-items: center;
+    border-style: solid;
+    padding: rem(16px 0);
```
Comment (LilBank) [inline src/styles/components/_sign-up-success-form.scss]: Let's remove this unused `padding`.

**Review by LilBank — CHANGES_REQUESTED** [review]

File: src/components/Form/index.tsx [hunk 25]
```diff
@@ -1,11 +1,19 @@
...
 interface Form {
   title: string
+  href: string
 }
 
-const Form = ({ title }: Form) => {
+const Form = ({ title, href }: Form) => {
+  const router = useRouter()
+
+  const onButtonSubmitClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (kimvcp) [inline src/components/Form/index.tsx]: How about this?
```suggestion
  const onSubmitButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```

**Review by kimvcp — APPROVED** [review]

**Review by LilBank — APPROVED** [review]

**Comment by LilBank** [comment]
Please edit these commit message.
